monai/fl/client/monai_algo.py

```diff
@@ -245,26 +267,46 @@ def get_weights(self, extra=None):
             extra: Dict with additional information that can be provided by FL system.
 
         Returns:
-            return_weights: `ExchangeObject` containing current weights.
+            return_weights: `ExchangeObject` containing current weights (default)
+            or "best" local weights if `extra` has `ExtraItems.MODEL_NAME` with string containing "best".
 
         """
         if extra is None:
             extra = {}
+
+        # by default return current weights, return best if requested via model name.
+        return_best = False
+        if ExtraItems.MODEL_NAME in extra:
+            if "best" in extra.get(ExtraItems.MODEL_NAME).lower():
```

Comment:
I feel it's not a robust way to determine the behavior by some string comparison.
Better to have a more clear argument in the `ExtraItems` in the `extra`, for example: `ExtraItems.MODEL_TYPE`.
@wyli What do you think?

Thanks.

Reply:
Changed to using `ModelType` for best and final model.